ligo/src/types.mligo

```diff
@@ -21,6 +21,17 @@ type ledger_key = address * token_id
 type ledger_value = nat
 type ledger = (ledger_key, ledger_value) big_map
 
+// Utility type to facilitate manipulation between internal functions
+// Note: it only has one "staked" as only past tokens can be staked
+type address_freeze_history =
```

Comment:
```suggestion
// Frozen token history for an address.
// This track the period number in which it was last updated and differentiates between
// tokens that were frozen during that period and the ones frozen in any other before.
// It does so because only tokens that were frozen in the past can be staked, which is 
// also why it tracks staked tokens in a single field.
type address_freeze_history =
```
A bit outdated/